lib/fs-monitor.ts

```diff
@@ -0,0 +1,41 @@
+import { EventEmitter } from 'events';
+import { FSWatcher } from 'fs';
+import { fs } from 'mz';
+import watch = require('node-watch');
+
+export interface FSEvent {
+	filename: string;
+	eventType: string;
+}
+
+export class FSMonitor extends EventEmitter {
+	private watcher: FSWatcher;
+
+	public constructor(private directory: string) {
+		super();
+	}
+
+	public async watch(): Promise<void> {
+		try {
+			await fs.access(this.directory, fs.constants.R_OK);
+		} catch (e) {
+			throw new Error(
+				`Could not access directory: ${this.directory} with error: ${e}`,
+			);
+		}
+
+		this.watcher = watch(
+			this.directory,
+			{ recursive: true },
+			(eventType, filename) => {
+				this.emit('fs-event', { filename, eventType });
+			},
+		);
+	}
+
+	public stopWatching(): void {
+		this.watcher.close();
```

Comment:
It looks like `this.watcher` is undefined until/unless `watch()` completes without raising an exception, right? It might be more friendly to users of this class to test `this.watcher` before closing it:
```
public stopWatching(): void {
	if (this.watcher) {
		this.watcher.close();
	}
}
```

Reply:
good spot

Reply:
Done, good catch.